### PR TITLE
chore(deps): Upgrade typescript to 4.7.4

### PR DESCRIPTION
**Pull request by jh2oman**
### Description

Almost the entire PR is handling errors from `try...catch` statements which are now type `unknown` but we often assume them to be type `Error`. I wrote a small helper function, `ensureError` to handle these cases.

4.7.X is the minimum version we need for typechain https://github.com/dethcrypto/TypeChain/tree/master/packages/target-ethers-v6

### Test plan

Updated all unit tests

### CODE REVIEW

File: test/RootStateSchema.json
Comment (satish-ravi) [inline test/RootStateSchema.json:1]: Are the changes here expected? Looks like we're losing a bunch of details from the schema

File: src/utils/Logger.ts [hunk 1]
```diff
@@ -53,10 +54,11 @@ class Logger {
   error = (
     tag: string,
     message: string,
-    error?: Error,
+    unknownError?: unknown,
```
Comment (satish-ravi) [inline src/utils/Logger.ts:57]: Are there still places where Logger is called without ensureError?

Comment (jh2oman) [reply src/utils/Logger.ts:57]: Yes, over 100

File: test/RootStateSchema.json
Comment (jh2oman) [reply test/RootStateSchema.json:1]: These are not expected. I will investigate more to try and figure out why and if it matters

File: src/utils/ensureError.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { ensureError } from 'src/utils/ensureError'
+
+describe('ensureError', () => {
```
Comment (jophish) [inline src/utils/ensureError.test.ts:3]: Can we add a test for when we for some reason can't stringify the value?

**Review by jophish — COMMENTED** [review]
Just a general question -- is this a common/normal pattern to use? Was it ever the case that we were throwing a non-`Error` value in the first place? I guess I'm unclear as to why we have "lost" this type guarantee. Is there another way to access thrown values as `Error` objects without using a psuedo-type-guard like this every time?

**Comment by jh2oman** [comment]
@jophish This is all necessary because previously catch(error) was type `any` and now it defaults to `unknown` as of typescript 4.4. more on that [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#defaulting-to-the-unknown-type-in-catch-variables---useunknownincatchvariables)

 I was inspired by [this article ](https://medium.com/with-orus/the-5-commandments-of-clean-error-handling-in-typescript-93a9cbdf1af5#:~:text=More%20importantly%3A%20before%20using%20a,block%20unknown%20instead%20of%20any%20.)for the ensureError solution. We would at minimum have to do something like `if(err instanceof Error)` everytime we access `err.message` but then what do we do in the else case? I thought it cleaner to add a helper like this.

we do in [at least one place](https://github.com/valora-inc/wallet/blob/main/src/identity/privateHashing.ts#L228) throw non-Error types on purpose

File: test/RootStateSchema.json [hunk 3]
```diff
@@ -312,9 +350,7 @@
                     "type": "string"
                 },
                 "category": {
-                    "enum": [
```
Comment (jh2oman) [inline test/RootStateSchema.json:315]: change made from this PR https://github.com/YousefED/typescript-json-schema/pull/535

File: test/RootStateSchema.json [hunk 4]
```diff
@@ -2,6 +2,45 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "additionalProperties": false,
     "definitions": {
+        "ActiveDapp": {
```
Comment (jh2oman) [inline test/RootStateSchema.json:5]: Changes all seem to be re-aranging or from [this enum to const change](https://github.com/YousefED/typescript-json-schema/pull/535)

File: test/RootStateSchema.json
Comment (jh2oman) [reply test/RootStateSchema.json:1]: I upgraded typescript-json-schema and now the changes look much more reasonable

File: src/account/reducer.ts [hunk 5]
```diff
@@ -18,15 +18,15 @@ export interface State {
...
+  startOnboardingTime?: number
   dismissedGetVerified: boolean
   dismissedGoldEducation: boolean
   acceptedTerms: boolean
   hasMigratedToNewBip39: boolean
-  choseToRestoreAccount: boolean | undefined
-  profileUploaded: boolean | undefined
-  recoveringFromStoreWipe: boolean | undefined
-  accountToRecoverFromStoreWipe: string | undefined
+  choseToRestoreAccount?: boolean
```
Comment (jh2oman) [inline src/account/reducer.ts:26]: new versions of ts-json-schema-generator do not recognize the `| undefined`

**Review by satish-ravi — APPROVED** [review]

File: src/account/reducer.ts (hunk 5 above)
Comment (jh2oman) [reply src/account/reducer.ts:26]: The rootstate should fail as long as the base state for the field is undefined

**Pull request merged**